src/bin/mdbook-xgettext.rs

```diff
@@ -109,12 +339,34 @@ fn main() -> anyhow::Result<()> {
         .ok_or_else(|| anyhow!("Missing output.xgettext.pot-file config value"))?
         .as_str()
         .ok_or_else(|| anyhow!("Expected a string for output.xgettext.pot-file"))?;
+    // The depth on which to split the output file. 0 (default) is to include
+    // all messages into a single POT file. Greater values will split POT files,
+    // digging into the sub-chapters within each chapter.
+    let split_depth = cfg
+        .get("depth")
+        .and_then(|value| value.as_integer())
+        .ok_or_else(|| anyhow!("Expected a number for output.xgettext.depth"))?;
     fs::create_dir_all(&ctx.destination)
         .with_context(|| format!("Could not create {}", ctx.destination.display()))?;
-    let output_path = ctx.destination.join(path);
-    let catalog = create_catalog(&ctx).context("Extracting messages")?;
-    polib::po_file::write(&catalog, &output_path)
-        .with_context(|| format!("Writing messages to {}", output_path.display()))?;
+    let catalogs = create_catalogs(&ctx, split_depth, path).context("Extracting messages")?;
+
+    for (file_rel_path, catalog) in catalogs {
+        let mut file_path = PathBuf::from(file_rel_path);
+        // Template files should have the "pot" extension.
+        assert!(file_path.set_extension("pot"));
+
+        let directory_path = {
```

Comment:
I think this is what [`parent()`](https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.parent) does?

Reply:
A large part of learning a new language is to figure out all the idioms and shortcuts :smile: